src/scenes/ModulesRouter/ModuleView.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { Box, Card, Grid, Typography, useTheme } from '@mui/material';
+import { Button } from '@terraware/web-components';
+
+import { Crumb } from 'src/components/BreadCrumbs';
+import Link from 'src/components/common/Link';
+import PageWithModuleTimeline from 'src/components/common/PageWithModuleTimeline';
+import { APP_PATHS } from 'src/constants';
+import useNavigateTo from 'src/hooks/useNavigateTo';
+import { useLocalization } from 'src/providers';
+import { requestGetModule } from 'src/redux/features/modules/modulesAsyncThunks';
+import { selectModule } from 'src/redux/features/modules/modulesSelectors';
+import { useAppDispatch, useAppSelector } from 'src/redux/store';
+import strings from 'src/strings';
+import { getLongDate, getLongDateTime } from 'src/utils/dateFormatter';
+
+import ModuleViewTitle from './ModuleViewTitle';
+
+const ModuleFieldDisplay = ({
+  label,
+  leftBorder = true,
+  onClickButton,
+  value,
+}: {
+  label: string;
+  leftBorder?: boolean;
+  onClickButton?: () => void;
+  value: string;
+}) => {
```

Comment:
Perhaps this should be extracted out into it's own file?

Reply:
Sounds like a good idea.